backend/main.go

```diff
@@ -30,6 +32,72 @@ type NewsApiResponse struct {
 	Articles     []Article `json:"articles"`
 }
 
+// POST Request Structure
+type ArticleRequest struct {
+	Keywords    string `json:"keyword" binding:"required"`
+	NumArticles int    `json:"numArticles" binding:"required"`
+	PhoneNumber string `json:"phoneNumber" binding:"required"`
+}
+
+type WhatsappMessage struct {
+	MessagingProduct string `json:"messaging_product"`
+	To               string `json:"to"`
+	Type             string `json:"type"`
+	Text             struct {
+		Body string `json:"body"`
+	} `json:"text"`
+}
+
+func sendMessageToWhatsapp(article Article, phoneNumber string) error {
+	whatsappToken := os.Getenv("WHATSAPP_TOKEN")
+	if whatsappToken == "" {
+		return fmt.Errorf("WhatsApp token is missing")
+	}
+	// メッセージ本文を作成
+	messageBody := fmt.Sprintf("Source: %s\nTitle: %s\nDescription: %s\nURL: %s\nPublished At: %s",
+		article.Source.Name, article.Title, article.Description, article.Url, article.PublishedAt)
+
+	// WhatsApp API用のメッセージ構造体を作成
+	message := WhatsappMessage{
+		MessagingProduct: "whatsapp",
+		To:               phoneNumber,
+		Type:             "text",
+	}
+	message.Text.Body = messageBody
+
+	// JSON形式にエンコード
+	messageData, err := json.Marshal(message)
+	if err != nil {
+		return fmt.Errorf("failed to marshal message: %v", err)
+	}
+
+	// WhatsApp APIにPOSTリクエストを送信
+	req, err := http.NewRequest("POST", "https://graph.facebook.com/v20.0/459929187208110/messages", bytes.NewBuffer(messageData))
```

Comment:
this should be env var too `459929187208110` (PHONE_NUMBER_ID). It is the number we send the message from.